Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmDirectoryHelper.cxx

```diff
@@ -182,7 +191,7 @@ std::string DirectoryHelper::RetrieveSOPInstanceUIDFromZPosition(double inZPos,
       DataElement de = itor->GetDataElement(thePosition);
       Attribute<0x0020,0x0032> at;
       at.SetFromDataElement( de );
-      if (fabs(at.GetValue(2) - inZPos)<0.01)
+      if (fabs(at.GetValue(2) - inZPos)<interSlice)
```

Comment:
Would it be possible to leave a comment explaining the role of `interSlicer` here? Just from looking at the code, and [the definition of `SpacingBetweenSlices`](http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.8.3.html#table_C.8-4) ("The spacing is measured from the center-to-center of each slice."), I cannot understand the logic.

Reply:
This was submited by @tbaudier see explanation at:

* https://github.com/malaterre/GDCM/pull/57

Reply:
@malaterre unrelated to this PR, but I am just curious, why don't you have `#define`s for DICOM attribute names, and always refer to them by tags (contrary to the conventions in DCMTK, for example). This practice strikes me as an approach that at the very least makes developers' life more complicated by requiring to always look up attribute names, and at extreme can cause errors by mis-typing tag numbers.

Reply:
Well we do have the mechanism since GDCM 2.4 (see Source/DataDictionary/gdcmTagKeywords.h).
For example, see some (rare) usage:
* https://github.com/malaterre/GDCM/blob/master/Source/MediaStorageAndFileFormat/gdcmEmptyMaskGenerator.cxx#L76-L77

So the only lame excuses I can write quickly:

1. Code using those keywords wont be compatible with GDCM 2.2
2. A totally made up excuse is that the header file is large, and I do not like waiting for compilation to finish
3. I should not do finger pointing, but a very well known and respected participant of WG6 promised that Keyword as defined in Part6 would never changed once introduced. However it already happen at least twice (top of my head) that keywords were updated (some kind of minor typos). This means code is fragile to DICOM editor workgroup change of mind.

Let me know if you want me to build other lame excuses not to start using those keywords in GDCM and make the codebase slightly easier to read ;)

Reply:
@malaterre thanks, that makes sense. I am glad that I did not hear hard resistance! I also saw this practice in ITK DCMTK IO module, and it really made me cringe ... Glad to hear the use of tags is not the convention that is encouraged or set in stone, even in GDCM proper!